spec/controllers/admin/users_controller_spec.rb

```diff
@@ -3,7 +3,9 @@
 require 'rails_helper'
 
 RSpec.describe Api::V1::Admin::UsersController, type: :controller do
-  let(:user) { create(:user) }
+  let(:role) { create(:role) }
+  let(:user) { create(:user, role:) }
+  let(:manage_users_permission) { create(:permission, name: 'ManageUsers') }
```

Comment:
@farhatahmad 
Following our call, I tried to add
`let(:role_permission) { create(:role_permission, permission: manage_users_permission, role:, value: 'true', provider: 'greenlight') }` but it would not work. I tried to debug but I had to let it go. It has been added to every endpoint the role_permission is needed.

Reply:
This should be possible :
` create(:role_permission, role_id: user.role.id, permission_id: permission.id, value: 'true', provider: 'greenlight')`.

This way we won't need to add the role_permission every test, but only to the test where you need to change the value to 'false'

Reply:
It doesn't seem to work. It should tho.

![image](https://user-images.githubusercontent.com/43917914/183701745-355a33fd-a62d-49e4-b5c0-d6b78bf160ac.png)

Reply:
Are you adding it in the before do block?

Reply:
No, I put it on top of everything, like the other let(:<something>)

![image](https://user-images.githubusercontent.com/43917914/183702854-8807dcdb-585f-4a61-8f94-d31b68c3f055.png)

Reply:
So I just learnt the let(:something) does not instantiate the object if the variable is not called.

Reply:
To force the initiation, use let!.

I think rubocop complains tho when using let!, that's why play around with it and see, but I think it has to be in before do

Reply:
let! works, thanks for the help.